tests/unit/gtBSMetadata.cpp

```diff
@@ -0,0 +1,167 @@
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";\
+	result = read_gbs_metadata(bitstream_guid_invalid3, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+
+	// valid input bitstream
+	uint8_t bitstream_guid_invalid4[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\":\
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\": [{\"total-contexts\":1,\
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid4, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+
+	// Invalid metadata afu-image node
+	uint8_t bitstream_guid_invalid5[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640 }, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid5, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+
+	// Invalid metadata interface-uuid
+	uint8_t bitstream_guid_invalid6[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\": \
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156,\
+		\"power\": 50,  \"magic-no\": 488605312, \"accelerator-clusters\": \
+		[{\"total-contexts\": 1, \"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid6, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+
+	// Invalid metadata afu-uuid
+	uint8_t bitstream_guid_invalid7[] = "XeonFPGA·GBSv001\53\02\00\00{\"version\": 640, \"afu-image\": \
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\":\
+		[{\"total-contexts\": 1, \"name\": \"nlb_400\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid7, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid8[] = "XeonFPGA·GBSv001\00\00\00\00{\"version\": 640, \"afu-image\":\
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156, \
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\": [{\"total-contexts\":1,\
+		\"name\": \"nlb_400\", \"accelerator-type-uuid\":\
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";
+	result = read_gbs_metadata(bitstream_guid_invalid8, &gbs_metadata);
+	EXPECT_NE(result, FPGA_OK);
+}
+
+/**
+* @test    bs_metadata_02
+* @brief   Tests: validate_bitstream_metadata
+* @details validate_bitstream_metadata validates BS metadata
+*          Retuns FPGA_OK if metadata is valid
+*/
+TEST(LibopaecBSMetadataCommonMOCKHW, bs_metadata_02) {

```

Comment:
Why do we need one test for this? Why can't we move this expect statement up to the other test? Or, conversely,  why can't we separate the expect statements above into individual tests? For simplicity, I would opt for the former. @tswhison, what do you think?

Reply:
The test case is getting sortof long, but I would be happy to have the functionality in place for now. :)